apps/portal-api/src/thirdparty/hubspot/hubspot.ts

```diff
@@ -67,20 +67,37 @@ export const hubspotLoginHook = async (userId: string) =>
     };
   });
 
-export const hubspotReachOutSalesHook = async () =>
+export const hubspotReachOutSalesHook = async ({
+  platformToken,
+  userId,
+}: {
+  platformToken?: string;
+  userId?: string;
+}) =>
   hubspotHook('reachOutSales', async () => {
-    const deploymentRequest =
-      await DeploymentRequestDomain.loadProvisionedTrialDeploymentRequestByPlatformIdentifier(
-        PlatformIdentifier.Opencti
-      );
+    let deploymentRequest;
+    if (userId) {
+      deploymentRequest =
+        await DeploymentRequestDomain.loadProvisionedTrialDeploymentRequestByPlatformIdentifier(
+          PlatformIdentifier.Opencti,
+          userId
+        );
+    } else if (platformToken) {
+      deploymentRequest =
+        await DeploymentRequestDomain.loadProvisionedTrialDeploymentRequestByPlatformToken(
+          platformToken
+        );
+    } else {
+      throw new Error('Either userId or platformToken must be provided');
+    }
 
     return {
```

Comment:
`deploymentRequest` is possibly not defined there if the `hub_status` is neither `active` nor `expired` for example.

I think we should throw an error.